Add missing CAN{1,2}, LPUART{1,2} modules; indexify CSE_PRAM, ADC{0,1}.{SC1,R} registers

jfhansen opened the pull request:
S32K144 PAC generated with svd2rust v0.17.0. As stated in the svd2rust changelog, it generates far more compact and more generic code than previous versions.
I have made some modifications in the .svd file which includes, making CSE_PRAM EmbeddedRAM into an array. The same is done with ADC0 and ADC1 SC1 channel registers and Data Result Registers, R. CAN1 and CAN2 peripherals are also added as separate structs, so that these peripherals can be used. The same is done with LPUART1 and LPUART2.

tmplt commented:
Thanks for the PR; adding this to my TODO list. I'll go through it in the coming months.

tmplt commented:
Closes #1.

tmplt commented:
This commit does not fix the CAN polymorphism issue described in #1, correct?

Other than that things look good to me. Please squash all changes done to `Cargo.toml` and remove the `wip-` prefix from the crate name. Crate version can be set to 0.11.0, as the other patch versions you have worked on are not published.

jfhansen commented:
Yes, it is correct that the polymorphism is not fixed. 
Alright. I will do that.

jfhansen commented:
This pull request has been reopened, and the suggested changes have been made.

tmplt commented:
There are still commits in the history in which the crate name is `wip-s32k144` and where the version number is `0.11.1` or `0.11.2`. Can you squash these changes with 07767e7?

jfhansen committed:
Generated crate with svd2rust v0.17.0. Added CAN1, CAN2, LPUART1, LPUART2. Indexified DMAMUX Channel Config registers into a Register Array, made ADC0 and ADC1's SC1 and R registers indexable. Made CSE_PRAM EmbeddedRAM Word-addressable array. Updated Cargo.toml file and Changelog

jfhansen commented:
There you go. I squashed the earlier commits, rewrote the commit messages so that the commits with the crate name `wip-s32k144` and version numbers 0.11.1 and 0.11.2 are removed from the history.

tmplt commented:
Looks good to me. I'll merge as soon as the CI passes.

tmplt commented:
Something is up with the CI. Anything you can fix @kjetilkjeka?

tmplt commented:
I'll handle this in January.

tmplt commented:
> I'll handle this in January.

I didn't. This PR is on indefinite hold until further notice.
